### PR TITLE
feature/business_report_metrics

### PR DESCRIPTION
**Pull request by Blokh**
Create new metrics and show them over the metrics page
Line of business distribution - West
Running Reports Status (How many are "In Progress" and "Manual Review", "Resolved", "Terminated") - South

## Summary by CodeRabbit

- **New Features**
	- Introduced a new status, `failed`, to the business report lifecycle for improved reporting accuracy.
	- Added a new schema for tracking report statuses, enhancing the granularity of metrics.
	- Added new properties for MCC counts and report statuses in the metrics output.

- **Enhancements**
	- Updated metrics retrieval methods to streamline data access and improve performance.
	- Expanded the `getHomeMetrics` method to include report status counts, enhancing the metrics provided.
	- Modified the `PortfolioRiskStatistics` component to focus on MCC analysis, improving clarity and functionality.
	- Introduced dynamic styling for report statuses to enhance visual feedback in the user interface.

- **Bug Fixes**
	- Removed outdated report retrieval logic to simplify data return structure.

### CODE REVIEW

File: services/workflows-service/src/metrics/service/metrics.service.ts [hunk 1]
```diff
@@ -96,15 +94,22 @@ export class MetricsService {
...
-      reports: {
-        all: await this.metricsRepository.getReportsByRiskLevel(currentProjectId),
-        inProgress: await this.metricsRepository.getInProgressReportsByRiskLevel(currentProjectId),
-        approved: await this.metricsRepository.getApprovedBusinessesReportsByRiskLevel(
-          currentProjectId,
-        ),
+      mccCounts,
+      riskIndicators,
+      reportStatuses,
+      reportsRisks: {
```
Comment (alonp99) [inline services/workflows-service/src/metrics/service/metrics.service.ts:111]: "inProgress" its not used?

File: apps/backoffice-v2/src/pages/Statistics/Statistics.page.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const Statistics: FunctionComponent = () => {
       <h1 className={'pb-5 text-2xl font-bold'}>Statistics</h1>
       <div className={'flex flex-col space-y-8'}>
         <UserStatistics fullName={'John Doe'} />
-        <PortfolioRiskStatistics riskIndicators={data.riskIndicators} reports={data.reports} />
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/Statistics.page.tsx:24]: Let's only pass the data in use as props.

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx [hunk 3]
```diff
@@ -205,16 +139,93 @@ export const PortfolioRiskStatistics: FunctionComponent<{
               </div>
             );
           })}
+          <div
+            key={'Report Statuses'}
+            className={'col-span-full min-h-[13.125rem] rounded-xl bg-[#F6F6F6] p-2'}
+          >
+            <Card className={'flex h-full flex-col px-3'}>
+              <CardHeader className={'pb-1'}>{'Report Status'}</CardHeader>
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx]:
```suggestion
              <CardHeader className={'pb-1'}>Report Status</CardHeader>
```

File: services/workflows-service/src/metrics/service/metrics.service.ts (hunk 1 above)
Comment (Blokh) [reply services/workflows-service/src/metrics/service/metrics.service.ts:111]: it's a part of statuses

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx [hunk 4]
```diff
@@ -205,16 +139,93 @@ export const PortfolioRiskStatistics: FunctionComponent<{
               </div>
             );
           })}
+          <div
+            key={'Report Statuses'}
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx:143]: Remove the key if the component is not in a loop

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx [hunk 5]
```diff
@@ -205,16 +139,93 @@ export const PortfolioRiskStatistics: FunctionComponent<{
...
+            className={'col-span-full min-h-[13.125rem] rounded-xl bg-[#F6F6F6] p-2'}
+          >
+            <Card className={'flex h-full flex-col px-3'}>
+              <CardHeader className={'pb-1'}>Report Status</CardHeader>
+              <CardContent>
+                <p className={'mb-8 text-slate-400'}>{'Merchant monitoring statuses'}</p>
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx]: The pattern with the object is only for dynamic content or for using string templates.
```suggestion
                <p className={'mb-8 text-slate-400'}>Merchant monitoring statuses</p>
```

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx [hunk 6]
```diff
@@ -205,16 +139,93 @@ export const PortfolioRiskStatistics: FunctionComponent<{
...
+                      y={60}
+                      textAnchor="middle"
+                      dominantBaseline="middle"
+                      className={'text-xs'}
+                    >
+                      {'Reports'}
+                    </text>
+                    <Pie
+                      data={reportStatuses.map(({ status, count }) => ({
+                        name: `${titleCase(status)}`,
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx]:
```suggestion
                        name: titleCase(status),
```

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx [hunk 7]
```diff
@@ -229,22 +240,85 @@ export const PortfolioRiskStatistics: FunctionComponent<{
...
+                        <div className={'h-full'}>
+                          <div
+                            className={`rounded bg-blue-200 p-1 transition-all`}
+                            style={{
+                              width: `${widths[index]}%`,
+                            }}
+                          >
+                            {`${mcc} - ${mccDescription}`}
+                          </div>
+                          {/*<span className={'relative z-50 ms-4'}>{titleCase(name ?? '')}</span>*/}
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/PortfolioRiskStatistics.tsx]: Let's remove the commented out parts. 🙏🏼

File: apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/hooks/usePortfolioRiskStatisticsLogic/usePortfolioRiskStatisticsLogic.tsx [hunk 8]
```diff
@@ -44,27 +41,27 @@ export const usePortfolioRiskStatisticsLogic = ({
...
+    //   name: 'Approved Merchant Onboarding',
+    //   description: 'Risk levels of all approved merchant onboarding reports',
+    //   entityPlural: 'Cases',
+    //   riskLevels: {
+    //     low: reportsRisks?.approved?.low,
+    //     medium: reportsRisks?.approved?.medium,
+    //     high: reportsRisks?.approved?.high,
+    //     critical: reportsRisks?.approved?.critical,
+    //   },
+    // },
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Statistics/components/PortfolioRiskStatistics/hooks/usePortfolioRiskStatisticsLogic/usePortfolioRiskStatisticsLogic.tsx]: Remove commented out code. I'd keep it if it was something to remember or complex.

File: services/workflows-service/src/metrics/repository/metrics.repository.ts [hunk 9]
```diff
@@ -260,34 +260,99 @@ export class MetricsRepository {
...
+
+      return acc;
+    }, {} as Record<keyof typeof BusinessReportStatus, number>);
+
+    const statusOrder: Array<keyof typeof BusinessReportStatus> = [
+      'completed',
+      'in_progress',
+      'new',
+      'failed',
+    ];
```
Comment (Omri-Levy) [inline services/workflows-service/src/metrics/repository/metrics.repository.ts]:
```suggestion
    const statusOrder = [
      'completed',
      'in_progress',
      'new',
      'failed',
    ] satisfies Array<keyof typeof BusinessReportStatus>;
```

File: services/workflows-service/src/metrics/repository/metrics.repository.ts [hunk 10]
```diff
@@ -260,34 +260,99 @@ export class MetricsRepository {
...
+        const [mcc, mccDescription] = mccName.split(' - ');
+
+        if (!mcc || !mccDescription) {
+          return;
+        }
+
+        return {
+          mcc: parseInt(mcc),
+          count,
+          percentage: Number(((count / total) * 100).toFixed(2)),
```
Comment (Omri-Levy) [inline services/workflows-service/src/metrics/repository/metrics.repository.ts:352]: Isn't it already a number?
```suggestion
          percentage: ((count / total) * 100).toFixed(2),
```

File: services/workflows-service/src/metrics/schemas/home-metrics.schema.ts [hunk 11]
```diff
@@ -7,16 +7,36 @@ export const ReportsByRiskLevelSchema = Type.Object({
   critical: Type.Number(),
 });
 
+// export const ReportStatusesCountSchema = Type.Object({
+//   [BusinessReportStatus.new]: Type.Optional(Type.Number()),
+//   [BusinessReportStatus.failed]: Type.Optional(Type.Number()),
+//   [BusinessReportStatus.in_progress]: Type.Optional(Type.Number()),
+//   [BusinessReportStatus.completed]: Type.Optional(Type.Number()),
+// });
```
Comment (Omri-Levy) [inline services/workflows-service/src/metrics/schemas/home-metrics.schema.ts]: Remove?

File: services/workflows-service/src/metrics/service/metrics.service.ts [hunk 12]
```diff
@@ -96,15 +94,22 @@ export class MetricsService {
...
 
-  async getHomeMetrics(currentProjectId: TProjectId): Promise<Static<typeof HomeMetricsSchema>> {
+  async getHomeMetrics(currentProjectId: TProjectId) {
+    // TODO - Add to read replica
+    const riskIndicators = await this.metricsRepository.getRiskIndicators(currentProjectId);
+    const reportStatuses = await this.metricsRepository.getReportStatusesCount(currentProjectId);
+    const mccCounts = await this.metricsRepository.getReportMCCsCount(currentProjectId);
+    const allRiskLevels = await this.metricsRepository.getReportsByRiskLevel(currentProjectId);
+    const completedRiskLevels =
+      await this.metricsRepository.getApprovedBusinessesReportsByRiskLevel(currentProjectId);
```
Comment (Omri-Levy) [inline services/workflows-service/src/metrics/service/metrics.service.ts]: Promise.all?

**Review by Omri-Levy — APPROVED** [review]

File: services/workflows-service/src/metrics/repository/metrics.repository.ts (hunk 10 above)
Comment (Blokh) [reply services/workflows-service/src/metrics/repository/metrics.repository.ts:352]: to fixed makes it string, theefore number parsing it required

**Pull request merged**